modules/apps/segments/segments-web/src/main/resources/META-INF/resources/js/components/odata_query_builder/ODataQueryBuilder.es.js

```diff
@@ -35,7 +36,8 @@ class ODataQueryBuilder extends Component {
 			{
 				criteriaMap: newCriteriaMap,
 				query: buildQueryString([newCriteriaMap])
-			}
+			},
+			this.props.onChange
```

Comment:
Should you be passing back something through `onChange`? 

```js
() => this.props.onChange(newValue)
```

Reply:
Nothing was passed into this since this was taking the values in the form, which is why this was a callback to the setState so there wouldn't be a race condition. (Although I suppose there still could be if the form input doesn't update after the state changes...)